### PR TITLE
stanford-corenlp 4.2.0

### PR DESCRIPTION
**Pull request by mattmoore**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/stanford-corenlp.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class StanfordCorenlp < Formula
   desc "Java suite of core NLP tools"
   homepage "https://stanfordnlp.github.io/CoreNLP/"
-  url "https://nlp.stanford.edu/software/stanford-corenlp-full-2018-10-05.zip"
-  version "3.9.2"
-  sha256 "833f0f5413a33e7fbc98aeddcb80eb0a55b672f67417b8d956ed9c39abe8d26c"
+  url "https://nlp.stanford.edu/software/stanford-corenlp-4.2.0.zip"
+  version "4.2.0"
```
Comment (carlocab) [inline Formula/stanford-corenlp.rb:5]:
```suggestion
```
We can remove this now that the version can be parsed from the URL.

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @mattmoore.

File: Formula/stanford-corenlp.rb (hunk 1 above)
Comment (mattmoore) [reply Formula/stanford-corenlp.rb:5]: @carlocab Good to know! Thanks.

File: Formula/stanford-corenlp.rb [hunk 2]
```diff
@@ -1,9 +1,8 @@
 class StanfordCorenlp < Formula
   desc "Java suite of core NLP tools"
   homepage "https://stanfordnlp.github.io/CoreNLP/"
-  url "https://nlp.stanford.edu/software/stanford-corenlp-full-2018-10-05.zip"
-  version "3.9.2"
-  sha256 "833f0f5413a33e7fbc98aeddcb80eb0a55b672f67417b8d956ed9c39abe8d26c"
+  url "https://nlp.stanford.edu/software/stanford-corenlp-4.2.0.zip"
+  sha256 "984aff1c431311a7dd5b4f0cb3bec4f5ba81da37aecc0d020b270571aafc5385"
   revision 1
```
Comment (carlocab) [inline Formula/stanford-corenlp.rb:6]:
```suggestion
```
Sorry; missed this one. `revision` lines need to be removed at version bumps.

File: Formula/stanford-corenlp.rb [hunk 3]
```diff
@@ -18,7 +16,7 @@ def install
 
   test do
     (testpath/"test.txt").write("Stanford is a university, founded in 1891.")
-    system "#{bin}/corenlp.sh", "-annotators tokenize,ssplit,pos", "-file test.txt"
-    assert_predicate (testpath/"test.txt.xml"), :exist?
+    system "#{bin}/corenlp.sh", "-annotators tokenize,ssplit,pos", "-file test.txt", "-outputFormat json"
```
Comment (mattmoore) [inline Formula/stanford-corenlp.rb:19]: Explicitly adding output format for the file, as 4.2.0 requires this option for file output formats.

**Review by carlocab — APPROVED** [review]
Thanks, @mattmoore.

**Pull request merged**